paasta_tools/secret_tools.py

```diff
@@ -170,3 +170,94 @@ def decrypt_secret_environment_variables(
         )
     )
     return decrypted_secrets
+
+
+def decrypt_secret_volumes(
+    secret_provider_name: str,
+    secret_volumes_config: Dict[str, Any],
+    soa_dir: str,
+    service_name: str,
+    cluster_name: str,
+    secret_provider_kwargs: Dict[str, Any],
+) -> Dict[str, str]:
+    secret_volumes = {}
+    # The config might look one of two ways:
```

Comment:
it's a little unfortunate that the kubernetes_tools and secrets_tools functions are so similar, but i'm not really sure how to dedupe them without making the code more painful to work with :p

Reply:
Same. It feels like this the concept of getting a secret provider can solve this, but it will likely take quite a bit of refactoring that I am hesitant to delve into at the moment. Or maybe something like a partial function? Something like
 
```
>>> def foo(a, b, c):
...  print("foo function")
...
>>> def bar(b, c, d, e):
...  print("bar function")
...
>>> def the_logic(secret_volumes, the_function, service_name):
...  for secret_volume in secret_volumes:
...   the_function(service_name, secret_volume["secret_name"])
...
>>> from functools import partial
>>> the_foo_partial = partial(foo, "a")
>>> the_bar_partial = partial(bar, "a", "b")
>>> the_logic([{"secret_name": "the_secret"}], the_foo_partial, "the_service")
foo function
>>> the_logic([{"secret_name": "the_secret"}], the_bar_partial, "the_service")
bar function
```

I think I prefer to save the exploration of such an improvement for another day.

Reply:
yea, the painful solution i had in mind was using partial functions - definitely fine to leave this for another day :)

Reply:
yeah, actually I think the "correct" solution here is to add a new `SecretProvider` for kubernetes, and have secret_tools just use the dynamically dispatched instance (kubernetes or vault). I might play with that in the future, but it's going to be a bunch of fallout from that (unit tests, etc) so I'm wary to sign up for it at the moment :)